uk.co.compucorp.civicrm.hrcore/backstop_data/page-objects/ssp-la-my-leave.js

```diff
@@ -0,0 +1,44 @@
+var Promise = require('es6-promise').Promise;
+var page = require('./page');
+
+var currentMonth = (new Date()).getMonth() + 1;
+
+module.exports = (function () {
+
+  return page.extend({
+    /**
+     * Wait for current month to be visible
+     */
+    waitForCurrentMonthsVisibility: function () {
+      var casper = this.casper;
+
+      casper.then(function () {
+        //Wait for the current month spinner to be first visible then hidden
+        casper.waitUntilVisible('.panel.panel-default:nth-of-type(' + currentMonth + ') ng-include .chr_spinner', function () {
+          casper.waitWhileVisible('.panel.panel-default:nth-of-type(' + currentMonth + ') ng-include .chr_spinner');
+        });
+      });
+    },
+    /**
+     * Clear the months selected to show all months
+     */
+    showAllMonths: function () {
+      var casper = this.casper;
+      //If Current month is december then need to wait for November as it would be loaded last when all months load
+      var lastMonth = currentMonth === 12 ? 11 : 12;
+
+      casper.then(function () {
+        //Wait for the current month spinner to be first visible then hidden
+        casper.waitUntilVisible('.panel.panel-default:nth-of-type(' + currentMonth + ') ng-include .chr_spinner', function () {
+          casper.waitWhileVisible('.panel.panel-default:nth-of-type(' + currentMonth + ') ng-include .chr_spinner', function () {
```

Comment:
if you put `waitForCurrentMonthsVisibility` in `waitForReady` as in previous comments, the page will automatically use it everytime you `init` it, so you don't have to repeat this in every method